### ISSUE TITLE
Refactor `.rename_column` utility function

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#417]
In PR #390 we used functionality to rename a column of a `data.frame` using `dplyr`:

```r
data <- .rename_column(data, "pdate_lwr", pdate_lwr)
data <- .rename_column(data, "pdate_upr", pdate_upr)
data <- .rename_column(data, "sdate_lwr", sdate_lwr)
data <- .rename_column(data, "sdate_upr", sdate_upr)
data <- .rename_column(data, "obs_date", obs_date)
```

Where:

```r
.rename_column <- function(df, new, old) {
  are_char <- is.character(new) & is.character(old)
  if (are_char) {
    df <- dplyr::rename(df, !!new := !!old)
  }
  return(df)
}
```

This is an inefficient approach as `dplyr` copies the `data` each time. It's also not very nice to read. We should refactor it.

**Comment by athowes** [comment]
Attempted this in [47056eb](https://github.com/epinowcast/epidist/pull/390/commits/47056eb25aacac8e62ab1bbfd69ce77a6a8273d0)

**Comment by athowes** [comment]
Closed by #390.
